### PR TITLE
ROCm changes in grappler

### PR DESCRIPTION
**Pull request by ekuznetsov139**
This PR:
Enables several grappler unit tests for ROCm
Prevents grappler from fusing convolution with other ops on ROCm (since ROCm does not support fused convolution at this time)

### CODE REVIEW

File: tensorflow/core/grappler/optimizers/remapper_test.cc [hunk 1]
```diff
@@ -69,9 +69,9 @@ TEST_F(RemapperTest, FusedBatchNorm) {
 }
 
 TEST_F(RemapperTest, FusedBatchNormNCHW) {
-#if !GOOGLE_CUDA
+#if !(GOOGLE_CUDA || TENSORFLOW_USE_ROCM)
   GTEST_SKIP() << "CUDA is not enabled";
```
Comment (deven-amd) [inline tensorflow/core/grappler/optimizers/remapper_test.cc]: "Neither ROCm nor CUDA is enabled" ?

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer_test.cc [hunk 2]
```diff
@@ -3577,9 +3577,9 @@ TEST_F(TransposerTest, ReduceTransposerKeepDims) {
 }
 
 TEST_F(TransposerTest, ReduceTransposerValidAxisNode) {
-#if !GOOGLE_CUDA
+#if !(GOOGLE_CUDA || TENSORFLOW_USE_ROCM)
   GTEST_SKIP() << "CUDA is not enabled";
```
Comment (deven-amd) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer_test.cc]: same comment as below for updating the skip message.

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_test.cc [hunk 3]
```diff
@@ -434,9 +434,9 @@ TEST_F(GenericLayoutOptimizerTest, Conv2DDataFormatVecPermuteCollapse) {
 }
 
 TEST_F(GenericLayoutOptimizerTest, DoNotPruneNonAddedCancellableTransposes) {
-#if !GOOGLE_CUDA
+#if !(GOOGLE_CUDA || TENSORFLOW_USE_ROCM)
   GTEST_SKIP() << "CUDA is not enabled";
```
Comment (deven-amd) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_test.cc]: same comment as befow for updating the skip message.

File: tensorflow/core/grappler/devices.cc [hunk 4]
```diff
@@ -67,7 +67,7 @@ int GetNumAvailableGPUs(
 }
```
Comment (deven-amd) [inline tensorflow/core/grappler/devices.cc]: can we take this opportunity to fill out the #else block at line 59 with the ROCm equivalent implementation?

Comment (ekuznetsov139) [reply tensorflow/core/grappler/devices.cc]: Changed to report the number of GPUs, ignoring the "CUDA compute capability" argument.

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

**Comment by ekuznetsov139** [comment]
Submitted upstream as #36639

**Pull request merged**